src/index.keyboardNav.test.js

```diff
@@ -0,0 +1,132 @@
+import test from 'ava'
+import React from 'react'
+import { mount } from 'enzyme'
+import DropdownTreeSelect from './index'
+
+const node = (id, label) =>
+  ({ id, label, value: label })
+
+const tree = {
+  ...node('c1', 'ccc 1'),
+  children: [
+    {
+      ...node('c2', 'ccc 2'),
+      children: [node('a1', 'aaa 1'), node('a2', 'aaa 2')]
+    },
+    {
+      ...node('c3', 'ccc 3'),
+      children: [node('a3', 'aaa 3'), node('b1', 'bbb 1'), node('b2', 'bbb 2')]
+    }
+  ]
+}
+
+test('some key presses opens dropdown on keyboardNavigation', t => {
+  ['ArrowUp', 'ArrowDown', 'Home', 'PageUp', 'End', 'PageDown', 'a', 'B'].forEach(key => {
+    const wrapper = mount(<DropdownTreeSelect data={tree} enableKeyboardNavigation />)
+    wrapper.find('.search').simulate('keyDown', { key })
```

Comment:
Since I see a lot of `simulate` being used in tests, here's a fair warning if you aren't aware of it already - `simulate` doesn't simulate actual keyboard or browser events. Its fake su=imulation - it only wires of `<eventName> to an `on<eventName>` method/function in your component.

So if you want to actually test browser DOM events, Enzyme is not your tool. 

That said, I, however, tend to agree with Enzyme team's philosophy here - we should not be testing browser's functionality anyway; rather ensure that out handler does it job when called upon. And that can be done by simply calling `wrapper.instance().onXXX()` directly and asserting the expected state.

Reply:
I am aware. But now I test both that the handler is triggered and it that it does it's job. So not sure what difference it would make.

Reply:
> So not sure what difference it would make.

The difference is that we're not depending on a flaky API (`enzyme.simulate`). 

> I've used cypress and testcafe for some uitesting but I hardly think this issue is the one to introduce that. Too many things to cover with ci, what browsers etc.

With you 100%. That was more of a phase 2 thingy where we set up some real-world tests that mimic an actual user. I imagine those tests should outside of any components and should not be aware of the code at all. My thinking is that as we're introducing more features, we need more checks & balances to ensure stable releases.

Reply:
Unfortunely could not get the same result without simulate even if all the needed props should be passed in the event. I think it's a state-race (sending new keys too soon). Don't want to modify the code to have a callback etc just because I need the test to work.

I've commented out the tries if you want to give it a go.

Reply:
Did you have a go on this? Should remove the commented code either way.

Reply:
For some reason, I didn't even see your earlier comment. I think Github only bring you to the latest comment. If it works with `simulate` then let's go with it.